course6/toyapp-adam/util/keyboard.cc

```diff
@@ -26,10 +26,10 @@ void Keyboard::Release(KeyCode keyCode)
 
 void Keyboard::ClearChanges()
 {
-	for(KeyCode i = 0; i < NUM_KEYCODES; ++i)
+	for(auto& key: mKeys)
 	{
-		mKeys[i].pressed = 0;
-		mKeys[i].released = 0;
+		key.pressed = 0;
+		key.released = 0;
```

Comment:
They can't have the same value at the same time, right? if one is 0, then the other one must 1. Which makes one of them redundant. Which suggests you only need a simple bool.

Apart fro that, I think you should just remove this class entirely.

Reply:
They _can_ have the same value at the same time. Between logic updates, the user could have both pressed and released the key very quickly, and we don't want to miss that. The simulation could, if it wants, do X thing when key is pressed, Y when released. In this case I want it to do both.